File: app/src/main/java/com/doubean/ford/data/vo/GroupSearchResult.java
package com.doubean.ford.data.vo;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.List;

@Entity(tableName = "group_search_results")
public class GroupSearchResult extends ListResult {
    @PrimaryKey
    @NonNull
    public String query;

    public GroupSearchResult(@NonNull String query, @Nullable List<String> ids, int totalCount,
                             @Nullable Integer next) {
        super(ids, totalCount, next);
        this.query = query;
    }


}

File: app/src/main/java/com/doubean/ford/data/vo/GroupSearchResult.kt
package com.doubean.ford.data.vo

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "group_search_results")
class GroupSearchResult(
    @PrimaryKey val query: String, override val ids: List<String>, override val totalCount: Int,
    override val next: Int?
) : ListResult